File: app/src/main/java/com/example/adouble/calibrationtest/CalibrationSerializer.java
package com.example.adouble.calibrationtest;

import android.content.Context;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONTokener;

import java.io.BufferedReader;
import java.io.FileNotFoundException;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.io.OutputStream;
import java.io.OutputStreamWriter;
import java.io.Writer;
import java.util.ArrayList;

/**
 * Created by Double on 2017/4/25.
 */

public class CalibrationSerializer {

    private Context mContext;
    private String mFilename;

    public CalibrationSerializer(Context c, String f) {
        mContext = c;
        mFilename = f;
    }

    public ArrayList<Calibration> loadCalibration() throws IOException, JSONException {
        ArrayList<Calibration> calibrations = new ArrayList<Calibration>();
        BufferedReader reader = null;
        try {
            InputStream in = mContext.openFileInput(mFilename);
            reader = new BufferedReader(new InputStreamReader(in));
            StringBuilder jsonString = new StringBuilder();
            String line = null;
            while ((line = reader.readLine()) != null)
                jsonString.append(line);
            JSONArray array = (JSONArray) new JSONTokener(jsonString.toString()).nextValue();
            for (int i = 0; i < array.length(); i++) {
                calibrations.add(new Calibration(array.getJSONObject(i)));
            }
        } catch (FileNotFoundException e) {
            e.printStackTrace();
        } finally {
            if (reader != null)
                reader.close();
        }

        return calibrations;
    }

    public void saveCalibrations(ArrayList<Calibration> calibrations) throws JSONException, IOException{
        JSONArray array = new JSONArray();
        for (Calibration c : calibrations)
            array.put(c.toJSON());

        Writer writer = null;
        try {
            OutputStream out = mContext.openFileOutput(mFilename, Context.MODE_PRIVATE);
            writer = new OutputStreamWriter(out);
            writer.write(array.toString());
        } finally {
            if (writer != null)
                writer.close();
        }
    }
}

File: app/src/main/java/com/zjgsu/ai/calibrationtest/CalibrationSerializer.kt
package com.zjgsu.ai.calibrationtest

import android.content.Context
import org.json.JSONArray
import org.json.JSONObject
import org.json.JSONTokener
import java.io.*

/**
 * Created by Double on 30/07/2017.
 */

class CalibrationSerializer constructor(private var mContext: Context, private var mFilename: String){

    fun loadCalibration(): ArrayList<Calibration> {
        var calibrations: ArrayList<Calibration> = ArrayList()
        var reader: BufferedReader? = null
        try {
            var input = mContext.openFileInput(mFilename)
            reader = BufferedReader(InputStreamReader(input))
            var jsonString = StringBuilder()
            var line: String? = reader.readLine()
            while (line != null) {
                jsonString.append(line)
                line = reader.readLine()
            }
            var array: JSONArray = JSONTokener(jsonString.toString()).nextValue() as JSONArray
            for (i in 0 until array.length())
                calibrations.add(Calibration(array.getJSONObject(i)))
        } catch (e: FileNotFoundException) {
            e.printStackTrace()
        } finally {
            reader!!.close()
        }
        return calibrations
    }

    fun saveCalibrations(calibrations: ArrayList<Calibration>) {
        var array: JSONArray = JSONArray()
        for (c in calibrations)
            array.put(c.toJSON())
        var writer: Writer? = null
        try {
            var out:OutputStream = mContext.openFileOutput(mFilename, Context.MODE_PRIVATE)
            writer = OutputStreamWriter(out)
            writer.write(array.toString())
        } finally {
            writer!!.close()
        }
    }
}
